### PR TITLE
Marshallable

### PR DESCRIPTION
**Pull request by jarsen**
Changes `ObjectConvertible` to `Marshallable`, as well as adds `Unmarshallable` and `Remarshallable`

refs #8
refs #14

### CODE REVIEW

**Comment by bwhiteley** [comment]
The names are backwards. 
Marshallable == serializable and Unmarshallable == deserializable.

**Comment by brianmullen** [comment]
Since the names are backwards, `Remarshallable` doesn't make sense for updating. Maybe we combine the update method with `Unmarshallable` and add a default implementation that doesn't do anything?

**Comment by jarsen** [comment]
Oh noes. I wonder if that means A) I'm dumb, B) it's not intuitive enough, or C) people just need to learn which is which.

re: `Remarshallable`, I think I would prefer to have a separate protocol. But that could work.

**Comment by jarsen** [comment]
The problem I guess with having an update function on `Unmarshallable` with a default implementation is that people don't have to implement it, and it might do nothing. I could see this being confusing. I say we just come up with a better name. `UnmarshallUpdatable`? Something along those lines.

**Comment by brianmullen** [comment]
I don't know if I am a big fan of the name `Unmarshallable`. I think its the `Un` prefix as the first thing that pops into my head is undoing something. I know we are technically undoing the "serialized" data but in still seems confusing. What about changing the name to `Demarshallable`?

I am ok with a separate protocol for updating (I would prefer it). Just with the current name of `Remarshallable` didnt make sense.

**Comment by jarsen** [comment]
Right, I agree `Resmarshallable` no longer makes sense. I do find `Unmarshallable` more confusing as well—because it could mean it can not be marshalled. How about `Unmarshalling` and `Marshalling` instead?

**Comment by jarsen** [comment]
And `UnmarshalUpdating`?

**Comment by jarsen** [comment]
Also, for the sake of consistency with the framework name, I think I should change everything to me marshal with one "L". It looks like it is technically correct, but marshaling is just a weird word that can be spelled two ways (first sentence https://en.wikipedia.org/wiki/Marshalling_%28computer_science%29)

**Comment by jarsen** [comment]
Here's a funny thing, I renamed `Unmarshaling.value` to `Unmarshaling.unmarshall`, and in doing so I actually broke it without realizing. Still conforms to `ValueType` but gets the default implementation :astonished:

File: Marshal/Marshaling.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//     / .. \
+//    ()'  '()
+//
+//
+
+import Foundation
+
+public protocol Marshaling {
+    func marshal() -> Object
+}
```
Comment (bwhiteley) [inline Marshal/Marshaling.swift]: Small nit: no newline at end of file.

File: Marshal/UnmarshalUpdating.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+//     / .. \
+//    ()'  '()
+//
+//
+
+
+import Foundation
+
+
+public protocol UnmarshalUpdating {
```
Comment (bwhiteley) [inline Marshal/UnmarshalUpdating.swift:17]: Small nit: vertical whitespace is inconsistent with Marshaling.swift (above and below the `import`)

**Comment by bwhiteley** [comment]
:+1:

File: Marshal/UnmarshalUpdating.swift (hunk 2 above)
Comment (jarsen) [inline Marshal/UnmarshalUpdating.swift:17]: I actually have noticed this and it has been driving me nuts. Saving it for a different commit or PR

File: Marshal/Marshaling.swift (hunk 1 above)
Comment (jarsen) [inline Marshal/Marshaling.swift]: good catch.

**Comment by bwhiteley** [comment]
:+1:

**Comment by jarsen** [comment]
fixes #8 
fixes #14

**Pull request merged**